### PR TITLE
docs: enable Microsoft Teams Notifications

### LINKED ISSUE #14741: Notifications: Native MSFT Teams target
Our notifications currently support SMTP and Webhook as delivery targets. We want to add native support for Teams workspaces as many customers use this as their internal messaging platform.

The native Teams integration should allow administrators to direct notifications towards a Teams workspace. Users should receive notifications via an app in that workspace, not a single channel. This way they have individual control of their inbox.

This product work should include server flags to set the Teams target and a new option in the deployment notifications settings UI for the delivery target. This should be paired with adding docs explaining how to connect our notifications with a teams workspace.

The Slack + teams integration docs can be a single guide.

Related: https://github.com/coder/coder/issues/14740

> Note: We'll only be adding MSFT Teams and Slack as natively supported delivery methods.

### PR DESCRIPTION
**Pull request by mtojek**
Fixes: https://github.com/coder/coder/issues/14741
Preview: [link](https://github.com/coder/coder/blob/14741-teams/docs/admin/notifications/teams.md)

This PR adds a subpage to Coder docs with a guide on integrating Coder with MS Teams notifications.

Screenshot 2024-09-24 at 13 32 06

### CODE REVIEW

File: docs/admin/notifications/teams.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    			"text": "**@{replace(body('Parse_JSON')?['title'], '"', '\"')}**"
+    		},
+    		{
+    			"type": "TextBlock",
+    			"text": "@{replace(body('Parse_JSON')?['body'], '"', '\"')}",
+    			"wrap": true
+    		},
+    		{
+    			"type": "ActionSet",
+    			"actions": [@{replace(replace(join(body('Parse_JSON')?['payload']?['actions'], ','), '{', '{"type": "Action.OpenUrl",'), '"label"', '"title"')}]
```
Comment (johnstcn) [inline docs/admin/notifications/teams.md:120]: Can we add some explanation for why these `replace()` calls are neccessary?

**Review by johnstcn — APPROVED** [review]

File: docs/admin/notifications/teams.md (hunk 1 above)
Comment (mtojek) [reply docs/admin/notifications/teams.md:120]: I can add an explanation below the snippet, as I'm not quite sure if Teams Workflow likes inline comments...

Comment (mtojek) [reply docs/admin/notifications/teams.md:120]: Added 👍

**Review by DanielleMaywood — APPROVED** [review]
Instructions read well to me 👍

**Review by johnstcn — APPROVED** [review]

**Pull request merged**